builtin/providers/aws/resource_aws_codecommit_trigger_test.go

```diff
@@ -0,0 +1,160 @@
+)
+
+func TestAccAWSCodeCommitTrigger_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCodeCommitTriggerDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCodeCommitTrigger_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCodeCommitTriggerExists("aws_codecommit_trigger.test"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.#", "1"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSCodeCommitTrigger_withChanges(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCodeCommitTriggerDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCodeCommitTrigger_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCodeCommitRepositoryExists("aws_codecommit_trigger.test"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.794783505.name", "noname"),
+				),
+			},
+			resource.TestStep{
+				Config: testAccCodeCommitTrigger_withChanges,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCodeCommitRepositoryExists("aws_codecommit_trigger.test"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_codecommit_trigger.test", "trigger.807285978.name", "noname_withChanges"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckCodeCommitTriggerDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).codecommitconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_codecommit_trigger" {
+			continue
+		}
+
+		_, err := conn.GetRepositoryTriggers(&codecommit.GetRepositoryTriggersInput{
+			RepositoryName: aws.String(rs.Primary.ID),
+		})
+
+		if ae, ok := err.(awserr.Error); ok && ae.Code() == "RepositoryDoesNotExistException" {
+			continue
+		}
+		if err == nil {
+			return fmt.Errorf("Trigger still exists: %s", rs.Primary.ID)
+		}
+		return err
+	}
+
+	return nil
+}
+
+func testAccCheckCodeCommitTriggerExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		codecommitconn := testAccProvider.Meta().(*AWSClient).codecommitconn
+		out, err := codecommitconn.GetRepositoryTriggers(&codecommit.GetRepositoryTriggersInput{
+			RepositoryName: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		log.Printf("[INFO] TriggerMetadata %s", rs.Primary.ID)
+
+		if *out.Triggers[0].Name != rs.Primary.ID {
```

Comment:
Is out.Triggers[0].Name the name of the repository? Or is it the name of a trigger?